packages/babel-parser/src/plugins/flow.js

```diff
@@ -484,7 +484,17 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       return this.finishNode(node, "InterfaceDeclaration");
     }
 
+    checkNotUnderscore(word: string) {
+      if (word === "_") {
+        throw this.unexpected(
+          null,
+          "`_` is only allowed as a type argument to call or new",
+        );
+      }
+    }
+
     checkReservedType(word: string, startLoc: number) {
+      this.checkNotUnderscore(word);
```

Comment:
Can we remove this, I covered this case in #8965 by adding _ to the `primitiveTypes` array.